lib/problems.rb

```diff
@@ -1,9 +1,26 @@
 require_relative './stack.rb'
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(n)
 def balanced(string)
-  raise NotImplementedError, "Not implemented yet"
+  return false if string.length.odd?
+
```

Comment:
I suggest using a hash to match open parens/braces/brackets with closing brackets.

```ruby
matching_parens = {
  "}" => "{",
  ")" => "(",
  "]" => "[",
}
```